src/modules/hip/kernel/mel_filter_bank.hpp

```diff
@@ -0,0 +1,164 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ __forceinline__ float calcMel(float *input, int melBin,
+                                         float *weightsDown, int *intervals,
+                                         int fftStride, int fftShift,
+                                         float normFactor)
+{
+    float out = 0;
+    int fftbin = intervals[melBin];
+    int fftBinEnd = intervals[melBin + 1];
+
+    float *in =  input + fftbin * fftStride + fftShift;
+    for (; fftbin < fftBinEnd; fftbin++, in += fftStride) 
+    {
+        auto weightUp = float(1) - weightsDown[fftbin];
+        weightUp *= normFactor;
+        out += *in * weightUp;
+    }
+
+    fftBinEnd = intervals[melBin + 2];
+    in =  input + fftbin * fftStride + fftShift;
+
+    for (; fftbin < fftBinEnd; ++fftbin, in += fftStride) 
+    {
+        auto weightDown = weightsDown[fftbin];
+        weightDown *= normFactor;
+        out += *in * weightDown;
+    }
+
+    return out;
+}
+
+__global__ void mel_filter_bank_tensor(float *srcPtr,
+                                       uint2 srcStridesNH,
+                                       float *dstPtr,
+                                       uint2 dstStridesNH,
+                                       int *srcDimsTensor,
+                                       float maxFreq,
+                                       float minFreq,
+                                       int numFilter,
+                                       float sampleRate,
+                                       bool normalize,
+                                       float *normFactors,
```

Comment:
Please check if you can combine multiple params into single param

Reply:
removed some of the un necessary params